test/Psy/Test/VersionUpdater/CheckerTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of Psy Shell.
+ *
+ * (c) 2012-2015 Justin Hileman
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Psy\Test\VersionUpdater;
+
+use Psy\Shell;
+use Psy\VersionUpdater\Checker;
+
+class CheckerTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @dataProvider malformedResults
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage Unable to check for updates
+     * @param $input
+     */
+    public function testExceptionInvocation($input)
+    {
+        $checker = $this->getMockBuilder(Checker::class)
+            ->setMethods(['downloadFromGitHub'])
+            ->getMock();
+        $checker->expects($this->once())->method('downloadFromGitHub')->willReturn($input);
+        $checker->isLatest();
+    }
+
+    /**
+     * @dataProvider jsonResults
+     * @param $assertion
+     * @param $input
+     */
+    public function testDataSetResults($assertion, $input)
+    {
+        $checker = $this->getMockBuilder(Checker::class)
+            ->setMethods(['downloadFromGitHub'])
+            ->getMock();
+        $checker->expects($this->once())->method('downloadFromGitHub')->willReturn($input);
+        $this->assertSame($assertion, $checker->isLatest());
+    }
+
+    /**
+     * @return array
+     */
+    public function jsonResults()
+    {
+        return [
+            [false, json_decode('{"tag_name":"v9.0.0"}')],
```

Comment:
Here goes...

- PHP 5.5 is now officially dead for some weeks... php 5.3 is not receiving support for something like 5 (!!) years.
- Still the ramblings about the fact that's it up to lib développer to upgrade and thus promote new versions. Why bother when everything works nicely with the same setup ?
- no need to do itsy beetsy things to go around deprecated or now inexistant limitations of older versions.
-Once again, psysh is a **dev** tool... nowadays, you really have to be fucked up for not having at least 5.5 in your local machine.
- Awesome features in recent versions, such as traits, optimizations (see above), generators, awesome autoloading in composer since 5.6, constant resolved at execution, ... and so on

Sorry for the langage on some points (especially before the last one, in which i don't want to be offensive (Well okay maybe a little :D )), but there you go. :}